### PR TITLE
Add nagim parser

### PR DESCRIPTION
**Pull request by vladsaveliev**
A couple of additions into GenericParser:

1. `verbose=False` option to make the output more clear for big amounts of samples,
2. `skip_checking_gcs_objects=True` to skip all bucket object existence and size checks and md5 checks to make it work faster,
3. Error handling when manifest file has no data.

Added `parse_nagim.py`. The script has 2 commands: `transfer` and `parse`.

```
python scripts/parse_nagim.py transfer \
    --tmp-dir ~/tmp \
    --transfer-crams \
    --transfer-gvcfs
```

Transfers the CRAM and GVCF files, along with corresponding indices from a Terra workspace to the GCP upload bucket. Currently only works under a personal account, which works well for a test dataset, but might be harder for the US workspace on the full data. Ideally the Terra bucket can be shared with a service account, so the transfer
command can be submitted to Hail Batch (with `--use-batch`), however I haven't yet figured out how to shared with a service account.

```
python scripts/parse_nagim.py parse \
    nagim.samples_list.tsv \
    --tmp-dir ~/tmp \
    --skip-checking-objects \
    --confirm
```

Assuming CRAM and GVCFs are transferred, populates the sample metadata DB objects. The first parameter is a path to a TSV with 2 columns: sample IDs used in the NAGIM run, and project ID (in a format defined below in `PROJECT_ID_MAP`), which I got from Shyam.

### CODE REVIEW

File: sample_metadata/parser/generic_parser.py [hunk 1]
```diff
@@ -199,6 +205,10 @@ def parse_manifest(
             sample_id = self.get_sample_id(row)
             sample_map[sample_id].append(row)
 
+        if len(sample_map) == 0:
+            logger.error(f'No sample information read from the manifest')
```
Comment (illusional) [inline sample_metadata/parser/generic_parser.py:209]: I think this should raise an exception.

File: scripts/parse_nagim.py [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+SRC_BUCKETS_TEST = [
+    # Australian Terra workspace
+    'gs://fc-7d762f69-bb45-48df-901b-b3bcec656ee0/2232b739-5183-4935-bb84-452a631c31ea',
+    # The US Terra workspace
+    'gs://fc-bda68b2d-bed3-495f-a63c-29477968feff/1a9237ff-2e6e-4444-b67d-bd2715b8a156',
+]
+
+SRC_BUCKETS_MAIN = []
+
+assert all(b.startswith('gs://fc-') for b in SRC_BUCKETS_MAIN + SRC_BUCKETS_TEST)
```
Comment (illusional) [inline scripts/parse_nagim.py:59]: Could use a comment to say all terra buckets start with fc?

File: scripts/parse_nagim.py [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+        with open(sample_tsv_file) as f:
+            parser.parse_manifest(f, dry_run=dry_run)
+
+
+def _fix_ids(
+    samples_from_found_files: List[Sample],
+    sample_to_project_path: str,
+    prod: bool,
+) -> List[Sample]:
+    sapi = SampleApi()
```
Comment (illusional) [inline scripts/parse_nagim.py:242]: Docstring please

File: scripts/parse_nagim.py [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+            f' | gsutil -m cp -I {upload_bucket}/'
+        )
+        if hbatch:
+            _add_batch_job(hbatch, cmd, f'Transfer GVCFs from {bucket}')
+        else:
+            _call(cmd)
+
+
+def _add_batch_job(hbatch, cmd, job_name):
+    j = hbatch.new_job(job_name)
```
Comment (illusional) [inline scripts/parse_nagim.py:323]: Why default 32 cores?

File: scripts/parse_nagim.py [hunk 5]
```diff
@@ -0,0 +1,467 @@
...
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def get_sample_id(self, row: Dict[str, any]) -> str:
+        return row['ext_id']
+
+    def get_sequence_meta(self, sample_id: str, row: GroupedRow) -> Dict[str, any]:
+        sequence_meta = dict()
+        if 'gvcf' in row and row['gvcf'] and row['gvcf'] != '-':
```
Comment (illusional) [inline scripts/parse_nagim.py:357]: How does row['gvcf'] be set to "-"?

File: scripts/parse_nagim.py [hunk 6]
```diff
@@ -0,0 +1,467 @@
...
+        )
+    return samples
+
+
+cli.add_command(transfer)
+cli.add_command(parse)
+
+if __name__ == '__main__':
+    # pylint: disable=no-value-for-parameter
+    cli(obj={})  # pylint: disable=unexpected-keyword-arg
```
Comment (illusional) [inline scripts/parse_nagim.py:467]: Isn't this generally:
```suggestion
    cli()  # pylint: disable=unexpected-keyword-arg
```

**Review by illusional — APPROVED** [review]
A few constructive comments, but generally looks good

File: scripts/parse_nagim.py (hunk 6 above)
Comment (vladsaveliev) [reply scripts/parse_nagim.py:467]: Remaining artefact from sharing context between the commands

File: scripts/parse_nagim.py (hunk 4 above)
Comment (vladsaveliev) [reply scripts/parse_nagim.py:323]: The `-m` option for gsutil enables multi-threaded copy, I think the more cores the faster?

File: scripts/parse_nagim.py (hunk 3 above)
Comment (vladsaveliev) [reply scripts/parse_nagim.py:242]: Didn't want to duplicate the comment where the function is called, but will add

Comment (vladsaveliev) [reply scripts/parse_nagim.py:249]: Actually, split it into 2 methods, one to fix sample IDs, another to set project IDs.

**Pull request merged**